### PR TITLE
Gal 184/deploy UI build pipeline

### PR DESCRIPTION
**Pull request by msandhuNHS**
Changes:

1. deploy yaml role assume, bucket name and region set as vars

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,8 @@ sbom-spdx*.json
 # misc
 .DS_Store
 *.pem
+/gpush.sh
```
Comment (zainm97) [inline .gitignore:25]: can we take this out and put it in on a separate PR. Its not related to the ticket

File: Makefile [hunk 2]
```diff
@@ -3,10 +3,11 @@ include ./scripts/init.mk
 # This file contains hooks into the project configuration, test and build cycle
 # as automated steps to be executed on a workstation and in the CI/CD pipeline.
 
+# asdf-install \
```
Comment (zainm97) [inline Makefile:6]: can we add this to the separate PR mentioned in above comment

File: README.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ This repository template aims to foster a user-friendly development environment
 - Python
 - Node.js
 
-Make use of this repository template to expedite your project setup and enhance your productivity right from the get-go. Enjoy the advantage of having a well-structured, self-documented project that reduces overhead and increases focus on what truly matters - coding!
+Make use of this repository template to expedite your project setup and enhance your productivity right from the get-go. Enjoy the advantage of having a well-structured, self-documented project that reduces overhead and increases focus on what truly matters - coding!!!
```
Comment (zainm97) [inline README.md:13]: remove

File: gfeature.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+git checkout -b "gal-$1/$2" develop
```
Comment (zainm97) [inline gfeature.sh:1]: changes to be put in separate PR

File: gpush.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+git add .
```
Comment (zainm97) [inline gpush.sh:1]: changes to be put in separate PR

File: package.json [hunk 6]
```diff
@@ -9,6 +9,8 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "asdf": "^1.2.2",
```
Comment (zainm97) [inline package.json:12]: is asdf a depandacy we need to commit to the project? We needed to do a global install for the commit hooks but not sure we should be adding to the repo

**Review by zainm97 — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (msandhuNHS) [reply .gitignore:25]: Sounds good, will remove these bash files

File: .github/workflows/deploy-to-s3.yaml [hunk 7]
```diff
@@ -23,9 +20,9 @@ jobs:
...
-          role-to-assume: arn:aws:iam::136293001324:role/github-oidc-role
-          role-session-name: githubRoleSession
-          aws-region: ${{ env.AWS_REGION }}
+          role-to-assume: ${{ vars.ASSUME_ROLE_ARN }}
+          role-session-name: galleri-frontend-github-deploy-to-cloudfront
+          aws-region: ${{ vars.AWS_REGION }}
       - name: Copy files to S3 with the AWS CLI
         run: |
-          aws s3 sync ./out s3://gps-client/
+          aws s3 sync ./out s3://${{ vars.BUCKET_NAME }}/
```
Comment (zainm97) [inline .github/workflows/deploy-to-s3.yaml:28]: i cant see any local directory that is ./out in your branch. The files that already exist in the bucket looks like they come from different locations within your local repo

**Review by zainm97 — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-to-s3.yaml (hunk 7 above)
Comment (msandhuNHS) [reply .github/workflows/deploy-to-s3.yaml:28]: nextjs puts production files in .out/ directory and is generated at build

**Review by zainm97 — APPROVED** [review]
Looks good sir

**Review by LaithM96 — CHANGES_REQUESTED** [review]
Looks all good and checks all passed, can you please update your code with the comments in each code :)

**Pull request merged**